### PR TITLE
Ferrite.jl GSoC 2023

### PR DESCRIPTION
**Pull request by termi-official**
A bit late to the party, but here are is a page for Ferrite.jl, a finite element toolbox developed in Julia, following the call for GSoC mentoring. Maybe needs one iteration for fine tuning the page.

cc @koehlerson @fredrikekre @logankilpatrick

### CODE REVIEW

File: jsoc/gsoc/ferrite-fem.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Ferrite.jl - Finite Element Toolbox - Summer of Code
+
+[Ferrite.jl](https://github.com/ferrite-fem/Ferrite.jl) is a Julia package providing the basic building blocks to develop finite element simulations of partial differential equations.
+The package provides extensive examples to start from and is designed as a compromise between simplicity and generality, trying to map finite element concepts 1:1 with the code in a low-level .
+Ferrite is actively used in teaching finite element to students at several universities across different countries (e.g. Ruhr-University Bochum and Chalmers University of Technology).
+Further infrastructure is provided in the form of different mesh parsers and a Julia based visualizer called [FerriteViz.jl](https://github.com/Ferrite-FEM/FerriteViz.jl).
+
+Below we provide a four of potential project ideas in [Ferrite.jl](https://github.com/ferrite-fem/Ferrite.jl).
+However, interested students should feel free to explore ideas they are interested in.
```
Comment (fredrikekre) [inline jsoc/gsoc/ferrite-fem.md:9]:
```suggestion
However, interested students should feel free to explore ideas they are interested in. Please contact any of the mentors listed below, or join the `#ferrite-fem` channel on the Julia slack to discuss.
```

File: jsoc/gsoc/ferrite-fem.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+
+## Adaptive Mesh Refinement
+
+**Difficulty**: Medium-Hard
+
+**Project size**: 300-350 hours
+
+**Problem**: While [Ferrite.jl](https://github.com/ferrite-fem/Ferrite.jl) provides infrastructure for the most common tasks in finite element assembly, it lacks an adaptive mesh infrastructure for adaptive finite element technology.
+We started implementing a p4est type adaptive mesh refinement, which is a highly optimized data structure for this task, could not find time to finalize it.
```
Comment (fredrikekre) [inline jsoc/gsoc/ferrite-fem.md:20]:
```suggestion
A preliminary implementation of a p4est type adaptive mesh refinement have been started, but not finalized yet.
```

File: jsoc/gsoc/ferrite-fem.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Information about the existing implementation is summarized [here](https://github.com/Ferrite-FEM/Ferrite.jl/blob/mk/p4est/p4est-ferrite.md).
+
+**Minimum goal**: Finalize the basic p4est implementation as described in the original paper [1], either starting from the [existing branch](https://github.com/Ferrite-FEM/Ferrite.jl/tree/mk/p4est) (recommended) or from scratch together with a set of tests.
+
+**Extended goal**: Intersting extensions might be to implement the optimizations proposed by Tobin Isaac [2], anisotropic refinement as in p6est [3] or generalizations to other geometries as in t8code [4].
```
Comment (fredrikekre) [inline jsoc/gsoc/ferrite-fem.md:25]:
```suggestion
**Extended goal**: Interesting extensions might be to implement the optimizations proposed by Tobin Isaac [2], anisotropic refinement as in p6est [3] or generalizations to other geometries as in t8code [4].
```

File: jsoc/gsoc/ferrite-fem.md (hunk 2 above)
Comment (ViralBShah) [reply jsoc/gsoc/ferrite-fem.md:20]: "has been started"?

Comment (fredrikekre) [reply jsoc/gsoc/ferrite-fem.md:20]: Updated.

**Comment by logankilpatrick** [comment]
Thank you!

**Pull request merged**